assets/js/pagination.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const CURRENT_CONTINUATION = (new URL(document.location)).searchParams.get("continuation");
+const CONT_CACHE_KEY = `continuation_cache_${encodeURIComponent(window.location.pathname)}`;
+
+function get_data(){
+    return JSON.parse(sessionStorage.getItem(CONT_CACHE_KEY)) || [];
+}
+
+function save_data(){
+    const prev_data = get_data();
+    prev_data.push(CURRENT_CONTINUATION);
+
+    sessionStorage.setItem(CONT_CACHE_KEY, JSON.stringify(prev_data));
+}
+
+function button_press(){
+    let prev_data = get_data();
+    if (!prev_data.length) return null;
+
+    // Sanity check. Nowhere should the current continuation token exist in the cache 
+    // but it can happen when using the browser's back feature. As such we'd need to travel
+    // back to the point where the current continuation token first appears in order to 
+    // account for the rewind.
+    const conflict_at = prev_data.indexOf(CURRENT_CONTINUATION);
+    if (conflict_at != -1) {
+        prev_data.length = conflict_at;
+    }
+
+    const prev_ctoken = prev_data.pop();
+
+    // On the first page, the stored continuation token is null.
+    if (prev_ctoken === null) {
+        sessionStorage.removeItem(CONT_CACHE_KEY);
+        let url = window.location.href.split('?')[0];
+        let params = window.location.href.split('?')[1];
+        let url_params = new URLSearchParams(params);
```

Comment:
You can consolidate the two cases of these into a function. Maybe something like:


```js
// Some pseudo js code

// Method to set the current page's continuation token
// Removes the continuation parameter when a continuation token is not given
function set_continuation(prev_ctoken = null){
    let url = window.location.href.split('?')[0];
    let params = window.location.href.split('?')[1];
    let url_params = new URLSearchParams(params);

    if (continuation) {
        url_params.set("continuation", prev_ctoken);
    } else {
        url_params.delete('continuation');
    };
}

````

Reply:
Thank you for the suggestion!